<launch>
  <arg name="web3_http_provider" default="https://rinkeby.infura.io/v3/6396152fbf0d48f995bbb677c8e8baed" />
  <arg name="web3_ws_provider" default="wss://rinkeby.infura.io/ws" />
  <arg name="erc20_token" default="xrt.5.robonomics.eth"/>
  <arg name="factory_contract" default="factory.5.robonomics.eth" />
  <arg name="ens_contract" default="0x9D70a7ecf0834ca14969f676e54834F12eaE5E10" />
  <arg name="keyfile" default="/var/lib/liability/keyfile" />
  <arg name="keyfile_password_file" default="/var/lib/liability/keyfile-psk" />

  <group ns="eth">
    <node pkg="ethereum_common" type="erc20_node" name="erc20_token" respawn="true" output="screen">
      <param name="web3_http_provider" value="$(arg web3_http_provider)" />
      <param name="web3_ws_provider" value="$(arg web3_ws_provider)" />
      <param name="token_contract" value="$(arg erc20_token)" />
      <param name="ens_contract" value="$(arg ens_contract)"
             if="$(eval len(arg('ens_contract')) > 0)"/>
      <param name="factory_contract" value="factory.5.robonomics.eth" />
      <param name="keyfile" value="$(arg keyfile)" />
      <param name="keyfile_password_file" value="$(arg keyfile_password_file)" />
    </node>

    <node pkg="ethereum_common" type="eth_node" name="eth_node" respawn="true" output="screen">
      <param name="web3_http_provider" value="$(arg web3_http_provider)" />
      <param name="web3_ws_provider" value="$(arg web3_ws_provider)" />
      <param name="token_contract" value="$(arg erc20_token)" />
      <param name="ens_contract" value="$(arg ens_contract)"
             if="$(eval len(arg('ens_contract')) > 0)"/>
      <param name="keyfile" value="$(arg keyfile)" />
      <param name="keyfile_password_file" value="$(arg keyfile_password_file)" />
    </node>
  </group>
</launch>
